tests/zenko_tests/node_tests/backbeat/LifecycleUtility.js

```diff
@@ -0,0 +1,91 @@
+const async = require('async');
+
+const { scalityS3Client, awsS3Client } = require('../s3SDK');
+const ReplicationUtility = require('./ReplicationUtility');
+
+class LifecycleUtility extends ReplicationUtility {
+    constructor(s3) {
+        super(s3);
+    }
+
+    deleteAllObjects(bucketName, cb) {
+        this.s3.listObjects({ Bucket: bucketName }, (err, data) => {
+            if (err) {
+                return cb(err);
+            }
+            async.each(data.Contents, (item, next) => {
+                const { Key } = item;
+                this.s3.deleteObject({ Bucket: bucketName, Key }, next);
+            }, cb);
+        });
+    }
+
+    deleteBucket(bucketName, cb) {
+        async.waterfall([
+            next => this.deleteAllObjects(bucketName, next),
+            next => this.s3.deleteBucket({ Bucket: bucketName }, next),
+        ], cb);
+    }
+
+    putBucketLifecycleConfiguration(
+        srcBucket, keyPrefix, date, destLocation, cb) {
+            this.s3.putBucketLifecycleConfiguration({
+                Bucket: srcBucket,
+                LifecycleConfiguration: {
+                    Rules: [{
+                        Status: 'Enabled',
+                        Prefix: keyPrefix,
+                        Transitions: [
+                            {
+                                Date: date,
+                                StorageClass: destLocation,
+                            },
+                        ],
+                    }],
+                },
+            }, cb);
+        }
+
+    waitUntilTransitioned(bucketName, key, destLocation, cb) {
+        let shouldContinue;
+        return async.doWhilst(next =>
+            this.s3.headObject({
+                Bucket: bucketName,
+                Key: key,
+            }, (err, data) => {
+                if (err) {
+                    return next(err);
+                }
+                shouldContinue = data.StorageClass !== destLocation;
+                if (shouldContinue) {
+                    return setTimeout(next, 5000);
+                }
+                return next();
+            }),
+        () => shouldContinue, cb);
+    }
+
+    compareObjectsAWS(srcBucket, destBucket, key, destLocation, cb) {
+        return async.series([
+            next => this.waitUntilTransitioned(
+                srcBucket, key, destLocation, next),
+            next => this.getObject(srcBucket, key, next),
```

Comment:
IMO the GET should be done *before* the transition occurs, otherwise we're not testing that transition did not alter the data, we're only testing that a GET on cloudserver returns the same content than a GET directly on the cloud (so testing the GET abstraction layer).

Reply:
Thanks! Nice catch there.